### ISSUE TITLE
Design ideas

### ISSUE THREAD

**Issue by mratsim** [bluenote10/NimData#45]
Interesting article on data table design.

https://uxdesign.cc/design-better-data-tables-4ecc99d23356

Feel free to close.

**Comment by Vindaar** [comment]
While I don't disagree with the article in itself, it really only talks about things to improve the user experience of (and thus maybe the insights possibly gained from) the representation of data tables. 
But it doesn't really have anything to say about possible improvements to the underlying data tables (although even there I imagine you could improve the user experience).

Still, for the browser viewer of `NimData` (and any other data frame) these things sound valuable! Although I suppose those features can be easily built with some HTML + CSS and maybe JS in a data frame independent way by just receiving the table as JSON.

In the end though I remain convinced that tables themselves are a bad way to look at your data. That's what data visualization (and not table visualization :P) is for after all. Now having this kind of table viewer in the browser combined with a super nice interactive vega plot builder... That would be something.

**Comment by mratsim** [comment]
Oh yeah, that was just about the visualization.

I really want to do some Vega stuff but I also want to do a lot of low-level plumbing :/, too little time so much to do ...

**Comment by Vindaar** [comment]
> too little time so much to do ...

... yup

**Comment by bluenote10** [comment]
Yes interesting article. This is mainly relevant for my [tabloo](https://github.com/bluenote10/tabloo) project. The table visualization in NimData simply bakes everything (data + CSS + JS) into static HTML, which obviously does not scale. Tabloo fully decouples visualization front-end from the data back-end. Currently I only have a Python backend, but the idea was to add a Nim backend at some point as well.

> In the end though I remain convinced that tables themselves are a bad way to look at your data. 

I have a totally different experience :-). For me, seeing data visualized in tables (with the possibilities of filtering + sorting) is by far the quickest way to get a first understanding of data. For things like manging your mails, or browsing files I feel much more efficient when I have a table. I feel that tables mainly have a bad reputation due to "mobile first" design, but fortunately I'm not forced to analyzing my data on a smart phone ;).

**Comment by Vindaar** [comment]
> > In the end though I remain convinced that tables themselves are a bad way to look at your data.
> 
> I have a totally different experience :-). For me, seeing data visualized in tables (with the possibilities of filtering + sorting) is by far the quickest way to get a first understanding of data. For things like manging your mails, or browsing files I feel much more efficient when I have a table. I feel that tables mainly have a bad reputation due to "mobile first" design, but fortunately I'm not forced to analyzing my data on a smart phone ;).

That's very interesting. I probably have a skewed perception due to the kind of data I usually look at. It's mostly just numerical data, where information printed as strings is usually very hard to make out / often gives a wrong impression. Extracting min/max values from a table is about all I use it for.

Once you add filtering to the options - and not just sorting - I'd probably change my mind somewhat. But that's not something I've really encountered when looking at tabular data. Ok, I guess in a way when searching on websites, which return data in a tabular form - where I'd agree. Well, and of course when programmatically dealing with data frames and then echoing the table, but that's not what I had in mind when I said tables are bad way to represent data.

So I guess I should rephrase my statement and say it really depends on the data and the tools provided by the table visualizer.

**Comment by mratsim** [comment]
> > In the end though I remain convinced that tables themselves are a bad way to look at your data.
> 
> I have a totally different experience :-). For me, seeing data visualized in tables (with the possibilities of filtering + sorting) is by far the quickest way to get a first understanding of data. For things like manging your mails, or browsing files I feel much more efficient when I have a table. I feel that tables mainly have a bad reputation due to "mobile first" design, but fortunately I'm not forced to analyzing my data on a smart phone ;).

I have a mixed experience, I scan tables much quicker but I don't want to miss [dinosaurs](https://www.autodeskresearch.com/publications/samestats)

![image](https://user-images.githubusercontent.com/22738317/68545575-ae431680-03ce-11ea-94cb-c219409f1b3c.png)

![image](https://user-images.githubusercontent.com/22738317/68545687-a768d380-03cf-11ea-97e8-b6fd2bf8833e.png)


https://www.data-to-viz.com/caveats.html
https://www.geckoboard.com/learn/data-literacy/statistical-fallacies/
